README


This archive contains the distribution XML-Simple,
version 2.23:

  An API for simple XML files

This software is copyright (c) 2015 by Grant McLean.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.043.



Changes
Revision history for XML-Simple

2.23      2017-04-17 15:49:14+12:00 Pacific/Auckland
  - another fix to the logic for detecting circular references (RT#114120 from
    Matthias Pitzl, also report from Erland Sommarskog)

2.22      2015-12-05 11:08:20+13:00 Pacific/Auckland
  - POD fix in link to package XML::SAX::ParseFactory (from Mohammad S Anwar)
  - allow subclass to override attribute escaping (from Christopher Masto)
  - POD fix adding NAME section to FAQ (RT#110080 from Bart Martens & gregor
    herrmann of the Debian project)
  - POD typo fix (RT#110082 from gregor herrmann of the Debian project)
  - fix reference comparison operator used in checking for circular data
    structures (RT#89553 from Laurent Dami)
  - add better diagnostics from file handling code in test suite

2.21      2015-12-04 16:33:49+13:00 Pacific/Auckland
  - Improved diagnostics - xml_in() will now catch parse exceptions and add
    caller information via Carp::croak
  - Switch to use lexical warnings (suppress with either 'no warnings;' or
    'no warnings "XML::Simple";', rather than $^W = 0)
  - POD fixes (from David Steinbrunner)
  - Regex escaping fixes for Perl 5.22+ (from Kent Fredric)
  - Add workaround for test failures on RHEL/CentOS 6.5 (RT#102115 from
    Chris Drake, workaround from Tim Bunce)
  - Remove some 'eval' calls from test suite which could mask root cause of
    test failures
  - Make each test script use unique filenames to allow tests to be run in
    parallel (RT#101362 from Karen Etheridge & Kent Fredric)
  - POD update: mention XML::Twig in 'STATUS' section (RT#79228 from Matt Trout)

2.20      2012-06-20 22:00:13 Pacific/Auckland
  - Suppress warning from empty CDATA section (RT#58359 from Juan Antonio
    Navarro Pérez)

2.19_02   2012-06-19 20:33:53 Pacific/Auckland
  - Reinstate XML::SAX::Expat dependency
  - Fix warnings tests (broken in 2.19_01) which rely on global $^W

2.19_01   2012-06-17 23:27:22 Pacific/Auckland
  - Use lexical filehandles instead of old-style fileglobs (RT#77787 from
    Quanah)
  - Restrict scope of :strict mode to calling package (RT#41562 from Hannu
    Aronsson)
  - Fix POD typo (RT#45414 from Anirvan Chatterjee)
  - Update POD to reflect best practise regarding :strict and options
  - Change minimum Perl requirement to 5.008
  - Switch to using Dist::Zilla for release management

2.18  Aug 15 2007
  - Non-unique key attribute values now trigger a warning (or a fatal
    error in strict mode) rather than silently discarding data (patch
    from Daniel Baysinger)

2.17  Aug 02 2007
  - Added parse_string(), parse_file() and parse_fh() methods
  - Added default_config_file(), and build_simple_tree() hook methods
  - Tweak to implementation of exporting (patch from Stuart Moore)
  - Documented hook methods
  - Fixed test suite race condition (RT#28603 from Andreas J. König)

2.16  Oct 30 2006
  - Added test/fix for bad GroupTags option (report from Lee Goddard)
  - Added new_hashref() hook method
  - refactored cache save/restore methods for easier overriding

2.15  Oct 03 2006
  - Makefile.PL changes: reject known-bad PurePerl and RTF parser modules;
    default to XML::SAX::Expat if no parser installed
  - allow '.' characters in variable names (suggested by Cosimo Streppone)
  - fix output of undefs in arrayrefs with SuppressEmpty (reported by
    かんな - Kanna)
  - tidy up code and docs around lexical filehandle passed to OutputFile
    (report from Helge Sauer)
  - reduce memory usage by passing XML strings by reference (patch from
    Dan Sully)

2.14  Jan 29 2005
  - unlink and lock fixes for VMS (patch from Peter (Stig) Edwards)

2.13  Nov 17 2004
  - Fixed bug where NoIndent broke KeyAttr (reported by David Haas)
  - Added copy_hash helper method which may be overridden to avoid
    tied hashes becoming untied during XMLout (patch from Jan Sundberg)
  - Fixed bug where GroupTags corrupted source hashref in XMLout
    (reported by Bram)
  - Tweaks to SuppressEmpty for undef with XMLout (report from jamesb),
    behaviour now matches docs and additional behaviour of setting
    option to 1 will skip undefined values altogether

2.12  Apr 05 2004
  - added NumericEscape option
  - added ValueAttr option (patch from Anton Berezin)
  - suppress 'wide character in print' warning (reported by Dawei Lin)

2.11  Mar 02 2004
  - Fixed hash ordering assumption in a new test (reported by Jost Krieger)

2.10  Feb 29 2004
  - Added AttrIndent option (patch from Volker Moell)
  - Hash keys are now sorted alphabetically by default; enable the
    new NoSort option if you don't want this (patch from Volker Moell)
  - Fixed bug where disabling array folding broke anonymous array handling
  - Fixed bug when unfolding a tied hash
  - SuppressEmpty patch from Douglas Wilson
  - Numerous test improvements - Devel::Cover rocks!
  - POD update re XMLin(XMLout($data)) caveats (bug report from Slaven
    Rezic)

2.09  Sep 09 2003
  - Makefile.PL makeover contributed by Joshua Keroes
  - fixed hash ordering assumption in test script (reported by Michel
    Rodriguez)
  - POD updates
  - updated link to Perl XML FAQ

2.08  Jun 13 2003
  - fixed variable expansion not happening in attributes (patch from Paul
    Bussé)

2.07  May 20 2003
  - added test to catch old versions of Storable which lack locking support
  - removed new-style loop which broke on 5.005_03
  - suppress more uninitialised variable warnings

2.06  May 18 2003
  - fixed strict mode requiring ForceArray on output (fix from Igor Román
    Mariño)
  - fixed warnings about uninitialised values
  - minor POD update (link to FAQ)

2.05  Apr 16 2003
  - fixed warnings when NormaliseSpace undefined (reported by Peter
    Scott and others)
  - added support for specifying ForceArray using regular expressions
    (patch from Jim Cromie)
  - added check to escape_value to guard against undefined argument
    (reported by Henrik Gemal)
  - added NoIndent option (requested by Afroze Husain Zubairi)


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.043.
use strict;
use warnings;

use 5.008;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "An API for simple XML files",
  "AUTHOR" => "Grant McLean <grantm\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "XML-Simple",
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.008",
  "NAME" => "XML::Simple",
  "PREREQ_PM" => {
    "XML::NamespaceSupport" => "1.04",
    "XML::SAX" => "0.15",
    "XML::SAX::Expat" => 0
  },
  "TEST_REQUIRES" => {
    "Test::More" => "0.88"
  },
  "VERSION" => "2.23",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Test::More" => "0.88",
  "XML::NamespaceSupport" => "1.04",
  "XML::SAX" => "0.15",
  "XML::SAX::Expat" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
